src/api/v1/community_businesses/temporary/register.ts

```diff
@@ -0,0 +1,61 @@
+import * as Hapi from 'hapi';
+import * as Boom from 'boom';
+
+import { CommunityBusinesses, CbAdmins } from '../../../../models';
+import { isChildOrganisation } from '../../prerequisites';
+import { RegisterCommunityBusinessesRequest } from '../../types';
+import { response, cbPayload } from '../schema';
+
+import { RegionEnum, SectorEnum } from '../../../../models/types';
+
+export default [
+  {
+    method: 'POST',
+    path: '/community-businesses/register/temporary',
+    options: {
+      description: 'Register a temporary community businesses and admin user for onboarding',
+      auth: {
+        strategy: 'standard',
+        access: {
+          scope: ['organisations_details-child:write'],
+        },
+      },
+      validate: {
+        payload: {
+          orgName: cbPayload.name.required(),
+        },
+      },
+      pre: [
+        { method: isChildOrganisation, assign: 'isChild', failAction: 'error' },
+      ],
+      response: { schema: response },
+    },
+    handler: async (request: RegisterCommunityBusinessesRequest, h: Hapi.ResponseToolkit) => {
+      const {
+        pre: { isChild },
+        server: { app: { knex } },
+        payload: { orgName },
+      } = request;
+
+      if (!isChild) {
+        return Boom.forbidden('Insufficient permissions to create organisations');
+      }
+
+      // Create accounts
+      const { admin, cb } = await knex.transaction(async (trx) => {
+        const cb = await CommunityBusinesses.add(trx, {
+          name: `TEMPORARY ACCOUNT: ${orgName}`,
```

Comment:
I've just created two accounts with the same name by accidentally double clicking the button. I also created an account with the same name as an existing org.

So I think we should check for existence of orgs with the name `orgName` and `TEMPORARY ACCOUNT: ${orgName}`

Reply:
Why can't a CB have two demo accounts?

Reply:
@eliascodes I do not think it is necessary to restrict cbs to only one temp account. Let me know if you can think of any unintended side effects of this.